ansible/library/get_feature.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/python
+
+DOCUMENTATION = '''
+module:         get_feature
+version_added:  "1.0"
+author:         Yutong Zhang (yutongzhang@microsoft.com)
+short_description: Retrieve features status from DUT by 'show feature status' or 'show features' command 
+description:
+    - Retrieve features status from DUT by 'show feature status' or 'show features' command
+    - The retrieved features status will be returned as a dict 
+'''
+
+EXAMPLES = '''
+- name: Get feature status from DUT
+  get_feature:
+'''
+
+# Example of the output
+'''
+The input:
+$ show feature status
+Feature         State            AutoRestart     SetOwner
+--------------  ---------------  --------------  ----------
+bgp             enabled          enabled
+database        always_enabled   always_enabled
+
+The output:
+{
+    'bgp': 'enabled',
+    'database': 'always_enabled'
+}
+'''
+
+class FeatureModule(object):
+    def __init__(self):
+        self.module = AnsibleModule(argument_spec=dict())
+
+    def run(self):
+        """
+            Main method of the class
+        """
+
+        command_list = ['show feature status', 'show features']
+        for cmd in command_list:
+            try:
+                rc, out, err = self.module.run_command(cmd, executable='/bin/bash', use_unsafe_shell=True)
+            except Exception as e:
+                self.module.fail_json(msg=str(e))
```

Comment:
I assume your intention is resolve the compatibility issue of `show feature status` and `show features` commands on different versions.

If my assumption is correct, then there is logic error in this module.

If `show feature status` is not supported on the DUT, then this module will exit early. The `show feature` command will not have a chance to be executed.

Reply:
fix